src/App.tsx

```diff
@@ -8,53 +9,102 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUserPosts } from './api/posts';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsIsLoading, setPostsIsLoading] = useState(false);
+  const [hasPostsError, setHasPostsError] = useState('');
```

Comment:
The `hasPostsError` state is initialized as a string. Consider initializing it as a boolean or an object to allow for more flexible error handling in the future.